presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -426,7 +460,21 @@ public TimeBoundary(
 
     public TimeBoundary getTimeBoundaryForTable(String table)
     {
-        String responseBody = sendHttpGetToBroker(table, String.format(TIME_BOUNDARY_API_TEMPLATE, table));
-        return timeBoundaryJsonCodec.fromJson(responseBody);
+        try {
+            String responseBody = sendHttpGetToBroker(table, String.format(TIME_BOUNDARY_API_TEMPLATE, table));
+            return timeBoundaryJsonCodec.fromJson(responseBody);
+        }
+        catch (Exception e) {
+            if ((e instanceof PinotException)) {
+                // New Pinot broker will set response code 404 if time boundary is not set.
+                // This is backward incompatible with old version, as it returns an empty json object.
+                // In order to gracefully handle this, below check will extract response code and return empty json
+                // if not found time boundary.
+                if (e.getMessage().split(" ")[3].equalsIgnoreCase("404")) {
```

Comment:
make "404" a constant?
check and guard no array index out of bound exception when accessing [3]? could you please comment error message format

Reply:
Done